zirgen/circuit/predicates/gen_predicates.cpp

```diff
@@ -158,6 +158,21 @@ template <typename Func> void addSingleton(Module& module, const std::string nam
                     });
 }
 
+template <typename Func> void addUnion(Module& module, const std::string name, Func func) {
```

Comment:
This appears identical to addJoin.  I.e. maybe addJoin isn't a great name, but it's intended for any simple binary predicate that doesn't need any special nondet state...